pkg/mysqlcluster/orc_reconciliation.go

```diff
@@ -106,11 +196,35 @@ func (f *cFactory) updateStatusFromOrc(insts []orc.Instance) {
 			f.updateNodeCondition(host, api.NodeConditionReplicating, core.ConditionFalse)
 		}
 
-		if !node.ReadOnly {
-			f.updateNodeCondition(host, api.NodeConditionMaster, core.ConditionTrue)
+		f.updateNodeCondition(host, api.NodeConditionMaster, core.ConditionFalse)
+
+		isReadOnly = isReadOnly && node.ReadOnly
+
+		if node.ReadOnly == true {
+			f.updateNodeCondition(host, api.NodeConditionReadOnly, core.ConditionTrue)
 		} else {
-			f.updateNodeCondition(host, api.NodeConditionMaster, core.ConditionFalse)
+			f.updateNodeCondition(host, api.NodeConditionReadOnly, core.ConditionFalse)
 		}
+
+	}
+
+	if len(getMasterForNode(insts)) != 0 {
+		masterHostName, check := verifyMasterArray(getMasterForNode(insts))
```

Comment:
you can merge those two functions, you will always use them like this `verifyMasterArray(getMasterForNode(insts))` so you can make just one function:
```
func determinMasterFor(inst []orc.Instace) (string, err)
```